grakn-core/src/main/java/ai/grakn/graql/Query.java

```diff
@@ -67,4 +66,9 @@
      */
     @CheckReturnValue
     boolean isReadOnly();
+
+    /**
+     * Get the transaction associated with this query
+     */
+    Optional<? extends GraknTx> tx();
```

Comment:
Another thing I would like to see die. Either the transactions should be embedded completely or not. This middleground is just silly.

Reply:
Agreed!